src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -253,14 +253,27 @@ public void leaveToken(DetailAST ast) {
      * @return true if given class should be declared as final otherwise false
      */
     private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
-        return desc.isWithPrivateCtor()
+        return (desc.isWithPrivateCtor() || (!desc.isWithPrivateCtor() && isClassPrivate(desc)))
```

Comment:
please do us a favor rename `desc` to `classDesc`.

> isWithPrivateCtor

become weird, as it should be `isWithOnlyPrivateCtors`. Class can have few ctors, some of them public , some private. Final is only when class is With Only Private Ctors.

do you agree with me ?

Reply:
yes i agree with you it must be something like 
`isWithOnlyPrivateCtor` I think the renaming you suggest that can be require to not make confusion such like `(desc.isWithPrivateCtor && !desc.isWithNonPrivateCtor)`

Reply:
>please do us a favor rename desc to classDesc.

Please do

Reply:
Done